### ISSUE TITLE
Testing segmentation with segnet_basic_camvid

### ISSUE THREAD

**Issue by varunmurali1** [alexgkendall/caffe-segnet#54]
I have downloaded the segnet_basic_camvid caffemodel from the model zoo and ran the test_segmentation_camvid.py script and am getting very strange results. Fine-tuning from these weights also resulted in the layer seemingly learning that majority of pixels as a single label. I even tried fine-tuning the network from the vgg weights and am getting similar results. I ran the training in every case for at least 10,000 iterations with the accuracy reported around ~0.90. 

Attached are the images (ground truth, segnet_basic with no modifications, segnet_after_finetuning).

I am running out of ideas on improving these results. 

Attached is an image from the 
![gt](https://cloud.githubusercontent.com/assets/5446161/16547136/63f49580-4131-11e6-91d6-a8120ce23958.png)
![segnet_basic](https://cloud.githubusercontent.com/assets/5446161/16547135/63f33492-4131-11e6-9881-cfdcae809e4a.png)
![segnet_after_finetuning](https://cloud.githubusercontent.com/assets/5446161/16547134/63f21c9c-4131-11e6-9bf9-79fa9adfc4b8.png)

**Comment by varunmurali1** [comment]
I have now even tried different computers with different versions of CUDA and CUDNN to isolate this issue and it is repeatable across them. Failing to understand this performance.

**Comment by anilrgukt** [comment]
Dear Varun, 

Were you able to resolve the issue?

thanks,
Anil

**Comment by varunmurali1** [comment]
There were some issues with the testing script that I was using, there was a label mismatch between the CamVid dataset, and the web demo that threw me off. Also, I think I ran the batch normalization statistics for the CamVid dataset only and not the data I added.  Since, the error was on my end I closed the issue.

**Comment by anilrgukt** [comment]
Just a suggestion, when using the trained model(**.caffemodel) because of Batch normalization you have to first run Scripts/compute_bn_stat.py and then use the generated model for semantic segmentation. This worked for me.

thanks,

**Comment by sajjo79** [comment]
Hi,
I am using segnet basic model to preform segmentation on medical images. I tried the segnet as directed by the author but could not get the results. There are five classes in my data but model only predicts one class

**Comment by varunmurali1** [comment]
The only times I found the model predicting only one class when I had the class weights for the loss from the frequency balancing wrong.

**Comment by sajjo79** [comment]
Hi, varun. I have taken the same code as given by segnet and run it on my data that consits of brain images containing tumor of 5 types. I have changed the parameters like file sizes, file paths etc. No change in network other than no. of classes in last layers. When i run the network, for few early iterations, it shows class-wise accuracy among different classes however after 20-30 iterations, it shows only one class accuracy as 1 and 0 for others. What could be the reason....
